### ISSUE TITLE
Caste recycler works only once

### ISSUE THREAD

**Issue by IndusAryan** [recloudstream/cloudstream#563]
### Steps to reproduce

open app and swipe casts on movie or a series
it will swipe
now go to other movie or series it will open recommendations fragment.

### Expected behavior

caste should work everytime 

### Actual behavior

it doesn't work always,probably users needs to belong to a upper caste :) :harold:

### Cloudstream version and commit hash

Latest PreRelease

### Android version

A13

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to pre-release version **[Latest](https://github.com/recloudstream/cloudstream/releases)**.
- [X] If related to a provider, I have checked the site and it works, but not the app.
- [X] I will fill out all of the requested information in this form.

**Comment by Luna712** [comment]
Now it mostly works, except if you open the recommendations and swipe on the cast area, then close recommendations, it still breaks it, but going back and back in fixes it again.

**Comment by Luna712** [comment]
I think this can be closed now. It does seem to work 99% of the time now. Only once in a while it decides it doesn't want to work.

**Comment by Luna712** [comment]
Actually this does seem to still happen if I go to recommendation and click one then on the new result scrolling the cast recycler tries to open recommendations again...

**Comment by IndusAryan** [comment]
yes, it happens when you do stuff on recommendations, probably a lifecycle issue, that fragment is fucked and holy fkin shit, 1 year passed since i partially fixed this critical bug, i really came a long way
